src/object.c

```diff
@@ -211,6 +211,361 @@ robj *dupStringObject(const robj *o) {
     }
 }
 
+
+/* This is a helper function for the COPY command.
+ * Duplicate a list object, with the guarantee that the returned object
+ * has the same encoding as the original one.
+ * 
+ * The resulting object always has refcount set to 1 */
+robj *dupListObject(robj *o) {
+    robj *lobj;
+    long llen, start = 0;
+    char buf[64];
+
+    serverAssert(o->type == OBJ_LIST);
+
+    switch (o->encoding) {
+        case OBJ_ENCODING_QUICKLIST:
+            lobj = createQuicklistObject();
+            break;
+        default:
+            serverPanic("Wrong encoding.");
+            break;
+    }
+    quicklistSetOptions(lobj->ptr, server.list_max_ziplist_size, server.list_compress_depth);
+    llen = listTypeLength(o);
+    listTypeIterator *iter = listTypeInitIterator(o, start, LIST_TAIL);
+    while (llen--) {
```

Comment:
A quicklist is a linked list of ziplists, which are binary blobs. You should be able to copy each blob and string them together in a quicklist, it'll probably be much faster than this.